tests/phpunit/integration/Core/User_Surveys/REST_User_Surveys_ControllerTest.php

```diff
@@ -103,6 +117,145 @@ public function test_register_routes() {
 		$this->assertEquals( \WP_REST_Server::READABLE, $args[0]['methods'] );
 		$this->assertTrue( is_callable( $args[0]['callback'] ) );
 		$this->assertTrue( is_callable( $args[0]['permission_callback'] ) );
+
+		$args = $routes['survey']->get_args();
+		$this->assertEquals( 'core/user/data/survey', $routes['survey']->get_uri() );
+		$this->assertEquals( \WP_REST_Server::READABLE, $args[0]['methods'] );
+		$this->assertTrue( is_callable( $args[0]['callback'] ) );
+		$this->assertTrue( is_callable( $args[0]['permission_callback'] ) );
+	}
+
+	protected function send_request( $endpoint, $data, $response ) {
+		remove_all_filters( 'googlesitekit_rest_routes' );
+		$this->controller->register();
+		$this->register_rest_routes();
+
+		$this->fake_proxy_site_connection();
```

Comment:
The proxy connection would be better placed in `set_up` or if not applicable to all tests, in the specific test since it has wider implications than what is necessary to send a request.